sigar_linux.go

```diff
@@ -343,12 +347,95 @@ func (self *NetIfaceList) Get() error {
 	return err
 }
 
+var inodeMatcher1 = regexp.MustCompile(`\Asocket:\[(\d+)\]\z`)
+var inodeMatcher2 = regexp.MustCompile(`\A\[0000\]:(\d+)\z`)
+
+func extractInode(linkName string) (uint64, error) {
+	results := inodeMatcher1.FindStringSubmatch(linkName)
+	// the inode number is held in the second result
+	if len(results) >= 2 {
+		inodeStr := results[1]
+		return strconv.ParseUint(inodeStr, 10, 64)
+	}
+	results = inodeMatcher2.FindStringSubmatch(linkName)
+	if len(results) >= 2 {
+		inodeStr := results[1]
+		return strconv.ParseUint(inodeStr, 10, 64)
+	}
+
+	return 0, errors.New("Failed to parse socket inode")
+}
+
+// Map pid to process name
+func buildPidMap(pids []int) map[int]string {
+	pidMap := make(map[int]string)
+	procState := ProcState{}
+	for _, pid := range pids {
+		err := procState.Get(pid)
+		if err == nil {
+			pidMap[pid] = procState.Name
+		}
+	}
+	return pidMap
+}
+
+func populatePidProcessName(netConnsPtr *[]NetConn) {
+	// Gather the list of pids
+	pids := ProcList{}
+	err := pids.Get()
+	if err != nil {
+		return
+	}
+
+	// For each process, read the links under the `fd` dir. If the linkName matches a specific form,
+	// the inode number is embedded in the name. See netstat source:
+	// https://github.com/ecki/net-tools/blob/master/netstat.c#L349
+	// Any discovered inodes are paired with the pid to match below
+	inodeCache := make(map[uint64]int)
```

Comment:
Should the map's value type be more specific than `int`, like `int64`?  I forget if it's good practice to always include the int size

Reply:
sigar <a href="https://github.com/scalingdata/gosigar/blob/master/sigar_interface.go#L342">already uses</a> `int` to describe pids, so partly I'm matching what's there. But `int` makes sense to me, because that would allow the size to shrink or expand based on the OS/arch.